<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="maximumSize">
       <size>
        <width>300</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="Laser">
       <attribute name="title">
        <string>Laser</string>
       </attribute>
      </widget>
      <widget class="QWidget" name="Motion">
       <attribute name="title">
        <string>Motion</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>HWP</string>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="2" column="1">
            <widget class="QDoubleSpinBox" name="HWP_rel_step">
             <property name="value">
              <double>5.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QPushButton" name="HWP_negative_step">
             <property name="text">
              <string>&lt;&lt;</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QDoubleSpinBox" name="HWP_move_to_angle">
             <property name="decimals">
              <number>4</number>
             </property>
             <property name="maximum">
              <double>360.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QPushButton" name="HWP_go">
             <property name="text">
              <string>Go</string>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QPushButton" name="HWP_positive_step">
             <property name="text">
              <string>&gt;&gt;</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0" colspan="3">
            <widget class="QLabel" name="HWP_angle">
             <property name="font">
              <font>
               <pointsize>28</pointsize>
              </font>
             </property>
             <property name="text">
              <string>nan</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QPushButton" name="HWP_go_home">
             <property name="text">
              <string>Home</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>PI</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="0" column="2">
            <widget class="QDoubleSpinBox" name="Pi_X_move_to">
             <property name="decimals">
              <number>3</number>
             </property>
             <property name="maximum">
              <double>100.000000000000000</double>
             </property>
             <property name="value">
              <double>50.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="1" column="3">
            <widget class="QToolButton" name="Pi_Y_go">
             <property name="text">
              <string>Go</string>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QDoubleSpinBox" name="Pi_Z_move_to">
             <property name="decimals">
              <number>3</number>
             </property>
             <property name="maximum">
              <double>100.000000000000000</double>
             </property>
             <property name="value">
              <double>50.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="2" column="3">
            <widget class="QToolButton" name="Pi_Z_go">
             <property name="text">
              <string>Go</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QCheckBox" name="Pi_X_On">
             <property name="text">
              <string>X</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QCheckBox" name="Pi_Z_On">
             <property name="text">
              <string>Z</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="Pi_XPos">
             <property name="text">
              <string>nan</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="Pi_YPos">
             <property name="text">
              <string>nan</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="Pi_ZPos">
             <property name="text">
              <string>nan</string>
             </property>
            </widget>
           </item>
           <item row="0" column="3">
            <widget class="QToolButton" name="Pi_X_go">
             <property name="text">
              <string>Go</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QDoubleSpinBox" name="Pi_Y_move_to">
             <property name="decimals">
              <number>3</number>
             </property>
             <property name="maximum">
              <double>100.000000000000000</double>
             </property>
             <property name="value">
              <double>50.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QCheckBox" name="Pi_Y_On">
             <property name="text">
              <string>Y</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QPushButton" name="Pi_XYZ_50mkm">
             <property name="text">
              <string>Center</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="Scan">
       <attribute name="title">
        <string>Scan</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QGroupBox" name="groupBox_3">
          <property name="title">
           <string>3D scan</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_3">
           <item row="1" column="0">
            <widget class="QLineEdit" name="scan3D_path">
             <property name="text">
              <string>./data/scan</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QToolButton" name="scan3D_path_dialog">
             <property name="text">
              <string>...</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0" colspan="2">
            <widget class="QTableWidget" name="scan3D_config">
             <property name="verticalScrollBarPolicy">
              <enum>Qt::ScrollBarAsNeeded</enum>
             </property>
             <property name="horizontalScrollBarPolicy">
              <enum>Qt::ScrollBarAsNeeded</enum>
             </property>
             <property name="autoScroll">
              <bool>true</bool>
             </property>
             <attribute name="horizontalHeaderDefaultSectionSize">
              <number>50</number>
             </attribute>
             <attribute name="horizontalHeaderHighlightSections">
              <bool>true</bool>
             </attribute>
             <attribute name="horizontalHeaderMinimumSectionSize">
              <number>10</number>
             </attribute>
             <attribute name="horizontalHeaderStretchLastSection">
              <bool>true</bool>
             </attribute>
             <attribute name="verticalHeaderDefaultSectionSize">
              <number>30</number>
             </attribute>
             <attribute name="verticalHeaderStretchLastSection">
              <bool>false</bool>
             </attribute>
             <row>
              <property name="text">
               <string>Z</string>
              </property>
             </row>
             <row>
              <property name="text">
               <string>Y</string>
              </property>
             </row>
             <row>
              <property name="text">
               <string>X</string>
              </property>
             </row>
             <column>
              <property name="text">
               <string>From</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>To</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Step</string>
              </property>
             </column>
             <item row="0" column="0">
              <property name="text">
               <string>0</string>
              </property>
             </item>
             <item row="0" column="1">
              <property name="text">
               <string>100</string>
              </property>
             </item>
             <item row="0" column="2">
              <property name="text">
               <string>2</string>
              </property>
             </item>
             <item row="1" column="0">
              <property name="text">
               <string>0</string>
              </property>
             </item>
             <item row="1" column="1">
              <property name="text">
               <string>100</string>
              </property>
             </item>
             <item row="1" column="2">
              <property name="text">
               <string>1</string>
              </property>
             </item>
             <item row="2" column="0">
              <property name="text">
               <string>0</string>
              </property>
             </item>
             <item row="2" column="1">
              <property name="text">
               <string>100</string>
              </property>
             </item>
             <item row="2" column="2">
              <property name="text">
               <string>1</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="2" column="0" colspan="2">
            <widget class="QPushButton" name="start3DScan">
             <property name="styleSheet">
              <string notr="true">background-color: rgb(0, 170, 0);</string>
             </property>
             <property name="text">
              <string>Start</string>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="data">
       <attribute name="title">
        <string>data</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QGroupBox" name="groupBox_4">
          <property name="title">
           <string>Thorlabs CCS200</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_4">
           <item row="0" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Integrate:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QDoubleSpinBox" name="usbSpectr_integr_time">
             <property name="decimals">
              <number>3</number>
             </property>
             <property name="minimum">
              <double>0.001000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.010000000000000</double>
             </property>
             <property name="value">
              <double>0.010000000000000</double>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QPushButton" name="usbSpectr_set_integr_time">
             <property name="text">
              <string>Set</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QSpinBox" name="usbSpectr_from">
             <property name="maximum">
              <number>3647</number>
             </property>
             <property name="value">
              <number>1400</number>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QSpinBox" name="usbSpectr_to">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>3648</number>
             </property>
             <property name="value">
              <number>1600</number>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Save:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QSpinBox" name="usbSpectr_save_from">
             <property name="maximum">
              <number>3647</number>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QSpinBox" name="usbSpectr_save_to">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>3648</number>
             </property>
             <property name="value">
              <number>3648</number>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_5">
          <property name="title">
           <string>DAQmx</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_5">
           <item row="0" column="0">
            <widget class="QCheckBox" name="DAQmx_preview">
             <property name="text">
              <string>preview</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QSpinBox" name="DAQmx_shift">
             <property name="maximum">
              <number>1000</number>
             </property>
             <property name="value">
              <number>12</number>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QPushButton" name="DAQmx_find_shift">
             <property name="text">
              <string>optimize</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="3">
            <layout class="QVBoxLayout" name="DAQmx_plot"/>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="startCalibr">
          <property name="text">
           <string>Calibrate</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <layout class="QVBoxLayout" name="main">
      <item>
       <layout class="QHBoxLayout" name="plotView"/>
      </item>
      <item>
       <layout class="QHBoxLayout" name="imageView"/>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionExit">
   <property name="text">
    <string>&amp;Exit</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
